hotshot-state-prover/src/service.rs

```diff
@@ -315,7 +368,37 @@ pub async fn submit_state_and_proof(
     // prepare the input the contract call and the tx itself
     let proof: ParsedPlonkProof = proof.into();
     let new_state: ParsedLightClientState = public_input.into();
-    let tx = contract.new_finalized_state(new_state.into(), proof.into());
+
+    let mut tx = contract.new_finalized_state(new_state.into(), proof.into());
+
+    // TODO: (alex) centralize this to a dedicated ethtxmanager with gas control
+    // frugal gas price: set to SafeGasPrice based on live price oracle
+    // safe to be included with low priority position in a block.
+    // ignore this if etherscan fail to return
+    if let Ok(res) =
+        reqwest::get("https://api.etherscan.io/api?module=gastracker&action=gasoracle").await
+    {
+        // parse etherscan response,
+        // if etherscan API changes, we log the warning and fallback to default (by falling through)
+        if let Ok(res) = res.json::<EtherscanPriceOracleResponse>().await {
+            tracing::info!("Etherscan gas oracle: {:?}", res.result);
+            match TryInto::<GasInfo>::try_into(res.result) {
+                Err(e) => {
+                    tracing::warn!("{}", e)
+                }
+                Ok(gas_info) => {
+                    if let TypedTransaction::Eip1559(inner) = &mut tx.tx {
+                        inner.max_fee_per_gas = Some(gas_info.safe_max_fee);
+                        inner.max_priority_fee_per_gas =
+                            Some(gas_info.safe_max_fee - gas_info.base_fee);
+                        tracing::info!("Setting maxFeePerGas to: {}", gas_info.safe_max_fee);
+                    }
+                }
+            }
+        } else {
+            tracing::warn!("!! Etherscan Price Oracle API changed !!")
```

Comment:
Probably want to include the returned error in this message

Reply:
done in [db9ab26](https://github.com/EspressoSystems/espresso-sequencer/pull/2148/commits/db9ab26f48ec7b4f2f37faa0cc815d7f20c5ae6d)